File: fiducial markers/fiducialMarkers.cpp

using namespace cv;

int main()
{
    cv::Mat markerImage;
    cv::aruco::Dictionary dictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_250);
    cv::aruco::generateImageMarker(dictionary, 23, 200, markerImage, 1);
    cv::imwrite("marker23.png", markerImage);
}

File: fiducial markers/fiducialMarkers.py
import cv2 as cv
print(cv.__version__)

#Predefined dictionary
predefinedDictionary = cv.aruco.getPredefinedDictionary(cv.aruco.DICT_5X5_250)
id = 3
sidePixels = 200
borderBits = 1
markerImage = cv.aruco.generateImageMarker(predefinedDictionary, id, sidePixels, borderBits)

cv.imwrite("markerPredefined"+str(id)+".png", markerImage)


#Automatic dictionary
numberOfMarkers = 10
numberOfBits = 5
for i in range(numberOfMarkers):
    automaticDictionary = cv.aruco.extendDictionary(numberOfMarkers, numberOfBits)
    markerImage = cv.aruco.generateImageMarker(automaticDictionary, i, sidePixels, borderBits)

    cv.imwrite("markerAutomatic"+str(i)+".png", markerImage)
